tests/abi/tools/revng-ensure-rft-equivalence/Main.cpp

```diff
@@ -74,58 +74,134 @@ int main(int Argc, char *Argv[]) {
   if (EC)
     ExitOnError(llvm::createStringError(EC, EC.message()));
 
-  // Gather `RawFunctionType`s present in both the models based on their IDs.
+  // Introduce a function pair container to allow grouping them based on their
+  // `CustomName`.
   struct FunctionPair {
-    model::RawFunctionType *Left;
-    model::RawFunctionType *Right;
+    model::RawFunctionType *Left = nullptr;
+    model::RawFunctionType *Right = nullptr;
   };
-  std::map<model::Type::IDType, FunctionPair> Functions;
-  for (model::UpcastableType &LeftType : LeftModel->Model->Types())
-    if (auto *Left = llvm::dyn_cast<model::RawFunctionType>(LeftType.get()))
-      Functions[Left->ID()].Left = Left;
-  for (model::UpcastableType &RightType : RightModel->Model->Types())
-    if (auto *Right = llvm::dyn_cast<model::RawFunctionType>(RightType.get()))
-      Functions[Right->ID()].Right = Right;
-
-  // Ensure their stack arguments have the same ID.
-  //
-  // If two `RawFunctionType`s have the same ID but their stack argument structs
-  // do not - replace the ID of the "Right" struct to be that of the "Left" one.
+  struct TransparentComparator {
+    using is_transparent = std::true_type;
+
+    bool operator()(llvm::StringRef LHS, llvm::StringRef RHS) const {
+      return LHS < RHS;
+    }
+  };
+  std::map<model::Identifier, FunctionPair, TransparentComparator> Functions;
+
+  // Gather all the `RawFunctionType`s present in the first model.
+  // Make sure there are no duplicated names.
```

Comment:
This tool is already not very straightforward. Are you sure we need to check this?
If the model is valid, we cannot have duplicated `CustomName`s.

Let's strive to keep this tool small.

Reply:
I guess this is bad wording on my part. It's less about duplicates, and more about ensuring that EVERY tested function has a `CustomName` AND that both models in question have the exact same set of names in use.

Reply:
This was reworked and there's a new thread for it: https://github.com/revng/revng/pull/232#discussion_r1099965149

I'll close this one.